### PR TITLE
Provide a first draft of the conda-forge triqs_ctseg package

### PR DESCRIPTION
**Pull request by Wentzell**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by Wentzell** [comment]
I confirm

**Comment by Thoemi09** [comment]
I confirm

**Comment by Wentzell** [comment]
@conda-forge-admin, please lint

**Comment by Wentzell** [comment]
@conda-forge/help-python-c Please Review
We do understand are looking into the linux build failure and will address it.
However we do not understand the osx build issue happening in the dependency resolution.
Please let us know if you have any insights

**Comment by Thoemi09** [comment]
Here is the error message:
```
Encountered problems while solving:
  - package triqs-3.3.1-py310hdfd9bad_1 requires openmpi >=5.0.5,<6.0a0, but none of the providers can be installed

Could not solve for environment specs
The following packages are incompatible
├─ openmpi 4.*  is installable and it requires
│  └─ mpi 1.0 openmpi, which can be installed;
└─ triqs 3.3.*  is not installable because there are no viable options
   ├─ triqs 3.3.1 would require
   │  └─ mpich >=4.2.2,<5.0a0 , which requires
   │     └─ mpi 1.0 mpich, which conflicts with any installable versions previously reported;
   └─ triqs 3.3.1 would require
      └─ openmpi >=5.0.5,<6.0a0 , which conflicts with any installable versions previously reported.
```

**Comment by Wentzell** [comment]
> Here's what I've got...
> 
> For **recipes/triqs_ctseg/meta.yaml**:
> 
>     * The following maintainers have not yet confirmed that they are willing to be listed here: wentzell, thoemi09. Please ask them to comment on this PR if they are.

This is not correct, we have already confirmed.

**Comment by Wentzell** [comment]
> Here is the error message:
> 
> ```
> Encountered problems while solving:
>   - package triqs-3.3.1-py310hdfd9bad_1 requires openmpi >=5.0.5,<6.0a0, but none of the providers can be installed
> 
> Could not solve for environment specs
> The following packages are incompatible
> ├─ openmpi 4.*  is installable and it requires
> │  └─ mpi 1.0 openmpi, which can be installed;
> └─ triqs 3.3.*  is not installable because there are no viable options
>    ├─ triqs 3.3.1 would require
>    │  └─ mpich >=4.2.2,<5.0a0 , which requires
>    │     └─ mpi 1.0 mpich, which conflicts with any installable versions previously reported;
>    └─ triqs 3.3.1 would require
>       └─ openmpi >=5.0.5,<6.0a0 , which conflicts with any installable versions previously reported.
> ```

Let me note that this versioning issue above not show up for our almost identical feedstock
https://github.com/conda-forge/triqs_cthyb-feedstock

File: recipes/triqs_ctseg/conda_build_config.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+mpi:
+  - mpich
+  - openmpi
+mpich:
+  - '4'
+openmpi:
+  - '5'
```
Comment (mattwthompson) [inline recipes/triqs_ctseg/conda_build_config.yaml:7]: These lines might be the important difference between this and your other feedstock https://github.com/conda-forge/triqs_cthyb-feedstock/blob/7a4b637fa3928155cdeaebdb2c2a4f2a934376b0/recipe/conda_build_config.yaml

**Comment by mattwthompson** [comment]
One of your dependencies insists on openmpi 4 - if your `triqs` package can also be built against that, it'd make things easier

**Comment by Wentzell** [comment]
Thank @mattwthompson for the response. I don't think triqs insists explicitly on any particular openmpi version. C.f. https://github.com/conda-forge/triqs-feedstock/blob/main/recipe/conda_build_config.yaml

**Comment by Wentzell** [comment]
So I tried pinning the version for openmpi and mpich in my commit 9abf611, this seems to have made a difference.
OSX build was instead getting much further but running into a time-out.

**Comment by Wentzell** [comment]
@conda-forge-admin, please restart ci

**Comment by Wentzell** [comment]
This PR is ready for merge

File: recipes/triqs_ctseg/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+requirements:
+  build:
+    - python                                 # [build_platform != target_platform]
+    - cross-python_{{ target_platform }}     # [build_platform != target_platform]
+    - numpy                                  # [build_platform != target_platform]
+    - wget                                   # [target_platform == 'osx-arm64']
+    - libtool                                # [target_platform == 'osx-arm64']
+    - autoconf                               # [target_platform == 'osx-arm64']
+    - automake                               # [target_platform == 'osx-arm64']
+    - llvm-openmp                            # [target_platform == 'osx-arm64']
```
Comment (xhochy) [inline recipes/triqs_ctseg/meta.yaml:24]: Why are these specific to `osx-arm64` and not for all osx-* variants?

Comment (Wentzell) [reply recipes/triqs_ctseg/meta.yaml]: Thank you @xhochy for pointing this out, these lines remained from our osx-arm64 porting experiments and should be redundant now. Pushed the removal and awaiting CI results.

**Review by xhochy — APPROVED** [review]

**Pull request merged**